### PR TITLE
Refactor layouts and make content fragments

### LINKED ISSUE #117: Content and content-split fragments
**Is this a BUG REPORT or FEATURE REQUEST?**:
Actually more of a refactor

**What happened**:
Single and split layouts display content and those contents can be fragments. Them not being fragments causes issues such as two layouts with only a single part different, not being able to make nav a fragment with ease and two different loops to load fragments.

**What you expected to happen**:
Content and content split be two fragments and remove split layout.

### PR DESCRIPTION
**Pull request by mpourismaiel**
**What this PR does / why we need it**:
Make content fragments and remove split layout

**Which issue this PR fixes** *(optional - uncomment and add issue)*:
fixes #117

### CODE REVIEW

File: exampleSite/content/about-split/content-split.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
++++
+fragment = "content-split"
+#disabled = true
+date = "2016-09-07"
+weight = 100
+#background = ""
+title = "About Syna"
+subtitle = ""
+
+content = """
```
Comment (stp-ip) [inline exampleSite/content/about-split/content-split.md]: Content is easier to be done within the actual content section. Makes formatting for the user easier. Basically moving this to after `++++` and then calling it via .Content should work.

File: exampleSite/content/about-split/content-split.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+Etiam eget metus elit. Praesent turpis mi, malesuada nec libero a, cursus varius leo. In malesuada gravida est, nec maximus turpis posuere eu. Proin rutrum mattis lectus. Ut mi elit, tincidunt ut aliquet consequat, facilisis vitae arcu. Proin ut massa auctor, imperdiet sem ac, varius justo. Integer porta tellus quis eros sollicitudin mollis. Donec nulla sem, porttitor et velit vitae, tincidunt congue erat. Vivamus urna libero, viverra non pharetra eu, bibendum id arcu. In fermentum bibendum purus, vitae pharetra ipsum gravida quis. Nunc vitae congue diam. Maecenas pharetra orci vel magna fermentum, vitae auctor sapien consectetur. Nunc tempor erat vel nisi ullamcorper mollis. Sed et lectus accumsan, pharetra nunc scelerisque, finibus leo. Sed tempor eros nec nunc porta sodales.
+"""
+
+[sidebar]
+  title = "Sidebar"
+  content = """
+So much information
+Phone numbers
+Details
+Or even more
```
Comment (stp-ip) [inline exampleSite/content/about-split/content-split.md]: Without 2 trailing whitespaces will render it on a single line. Probably not what we want here.

File: exampleSite/content/about/content.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+#disabled = true
+date = "2017-10-05"
+weight = 100
+#background = ""
+
+title = "About Syna"
+subtitle = ""
+
+
+content="""
```
Comment (stp-ip) [inline exampleSite/content/about/content.md]: Same as with split.

File: exampleSite/content/nesting/content.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
++++
+fragment = "content-single"
+#disabled = true
+date = "2017-10-05"
+weight = 100
+#background = ""
+description = "Nested Page"
+
+content = """
```
Comment (stp-ip) [inline exampleSite/content/nesting/content.md]: Same as with split.

File: exampleSite/content/nesting/content.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
++++
+fragment = "content-single"
+#disabled = true
+date = "2017-10-05"
+weight = 100
+#background = ""
+description = "Nested Page"
```
Comment (stp-ip) [inline exampleSite/content/nesting/content.md]: Probably should be `title`?

File: layouts/partials/fragments/hero.html [hunk 6]
```diff
@@ -1,9 +1,8 @@
 {{ $hero := .self.Params }}
-{{ $dot := .dot }}
 {{ $name := .self.Name }}
 {{ $name := replace $name "/index.md" "" }}
 {{ $name := strings.TrimSuffix ".md" $name }}
-{{ $path := $dot.File.Path }}
+{{ $path := .self.File.Path }}
```
Comment (stp-ip) [inline layouts/partials/fragments/hero.html]: Haha totally missed that as a workable solution.

File: exampleSite/content/about-split/content-split.md (hunk 1 above)
Comment (stp-ip) [reply exampleSite/content/about-split/content-split.md]: If that doesn't work, it would make the whole interface for the user a pain for content.

Comment (mpourismaiel) [reply exampleSite/content/about-split/content-split.md]: Actually tried that solution and it didn't work. Entire resource scope did not contain the markdown that comes after frontmatter. No idea why that happens though. And I agree this is ridiculous but I have no idea how to ease the pain.

File: exampleSite/content/nesting/content.md (hunk 5 above)
Comment (mpourismaiel) [reply exampleSite/content/nesting/content.md]: It's the mirror of https://github.com/okkur/syna/pull/118/files#diff-fadb7b097ab29fdd0268523ca6c0e097L5

Comment (mpourismaiel) [reply exampleSite/content/nesting/content.md]: Should probably add title though.

File: layouts/partials/fragments/hero.html (hunk 6 above)
Comment (mpourismaiel) [reply layouts/partials/fragments/hero.html]: Yeah I was surprised to see $dot in the single layout.

File: exampleSite/content/about-split/content-split.md (hunk 1 above)
Comment (stp-ip) [reply exampleSite/content/about-split/content-split.md]: Damn will look into it.

File: exampleSite/content/nesting/content.md (hunk 5 above)
Comment (stp-ip) [reply exampleSite/content/nesting/content.md]: Yeah, but I think that was a leftover mistake I would assume. We don't use description at all so it probably is save to migrate that to title.

**Comment by stp-ip** [comment]
One thing that might be annoying for users is the need for a content file to duplicate the title.
True it is more explicit, but is a bit annoying. We can leave it for now and if it becomes a user issue later, try to find a workaround. Don't think there is an easy one so.

File: exampleSite/content/about-split/content-split.md [hunk 7]
```diff
@@ -4,10 +4,17 @@ fragment = "content-split"
...
 
-content = """
+[sidebar]
+  title = "Sidebar"
+  content = """
+So much information
+Phone numbers
+Details
+Or even more
+"""
```
Comment (stp-ip) [inline exampleSite/content/about-split/content-split.md]: Trailing whitespace ;)

File: exampleSite/content/about/content.md [hunk 8]
```diff
@@ -4,12 +4,8 @@ fragment = "content-single"
 date = "2017-10-05"
 weight = 100
 #background = ""
-
-title = "About Syna"
-subtitle = ""
-
-
-content="""
+subtitle = "asdfasdfsaf"
```
Comment (stp-ip) [inline exampleSite/content/about/content.md]: Not sure I agree with the choice for example data here ^^
Also title could be added or what was the reason to scrape it?

File: exampleSite/content/nesting/content.md [hunk 9]
```diff
@@ -4,9 +4,7 @@ fragment = "content-single"
 date = "2017-10-05"
 weight = 100
 #background = ""
-description = "Nested Page"
-
-content = """
```
Comment (stp-ip) [inline exampleSite/content/nesting/content.md]: Let's add commented and empty titles, just as easy of use for people setting something up:
```
#title = ""
#subtitle = ""
```

File: exampleSite/content/about/content.md (hunk 8 above)
Comment (mpourismaiel) [reply exampleSite/content/about/content.md]: LMFAO That was not supposed to be seen.
I'll readd title.

File: exampleSite/content/nesting/content.md (hunk 9 above)
Comment (mpourismaiel) [reply exampleSite/content/nesting/content.md]: I removed `# Nested page` and added it as title. Also commented all the subtitles.

**Comment by stp-ip** [comment]
/LGTM
After fixing the conflicts that is. @lauragift21 please review again. Might merge before an approve from your side as it's the base for a few other issues, but if you find any additional issues we can do a follow on PR.

**Pull request merged**